packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -59,14 +70,38 @@ export function hasOne(definition?: Object) {
   return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
 }
 
-/**
- * Decorator for hasMany
- * @param definition
- * @returns {(target:any, key:string)}
- */
-export function hasMany(definition?: Object) {
-  const rel = Object.assign({type: RelationType.hasMany}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function hasMany<T extends typeof Entity>(
+  targetModel: T,
+  definition?: Partial<HasManyDefinition>,
+) {
+  // todo(shimks): extract out common logic (such as @property.array) to
+  // @relation
+  return function(target: Object, key: string) {
+    property.array(targetModel)(target, key);
+
+    const defaultFkName = camelCase(target.constructor.name + '_id');
+    if (
+      (!definition || !definition.keyTo) &&
+      (!targetModel.definition ||
+        !targetModel.definition.properties ||
+        !targetModel.definition.properties[defaultFkName])
+    ) {
+      // note(shimks): should we also check for the existence of explicitly
+      // given foreign key name on the juggler definition?
+      throw new Error(
+        `foreign key ${defaultFkName} not found on ${
+          targetModel.name
+        } model's juggler definition`,
+      );
+    }
+    const meta = {type: RelationType.hasMany, keyTo: defaultFkName};
+    Object.assign(meta, definition);
```

Comment:
What's going to happen if `definition` has `{type: RelationType.<somethingElse>}`?